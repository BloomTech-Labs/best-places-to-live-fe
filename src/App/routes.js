import React from 'react';
import {Route} from 'react-router-dom';
import Home from '../Home';

const Routes = () => {
  return (
    <>
      <Route exact path='/' component={Home} />
      {/* <Route path='/templ' component={Template} /> */}
      {/* <Route path='/multi' component={MultiStepForm} /> */}
      {/* <Route path='/'  component={} />  */}
      {/* <Route path='/'  component={} />  */}
      {/* <Route path='/'  component={} />  */}
      {/* <Route path='/'  component={} />  */}
    </>
  );
};

export default Routes;
