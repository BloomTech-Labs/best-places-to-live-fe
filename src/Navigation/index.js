import React from 'react';
import { Link } from 'react-router-dom'
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';

const useStyles = makeStyles(theme => ({
    root: {
        flexGrow: 3,
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },

}));

export default function Navigation() {
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <AppBar position="static">
                <Toolbar>
                    <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
                        <MenuIcon />
                    </IconButton>

                    <Link to="/" style={{textDecoration:'none',color:'white'}}>
                        <Button color="inherit">Home</Button>
                    </Link>

                    <Link to="/topten" style={{textDecoration:'none',color:'white'}}>
                        <Button color="inherit">Top10</Button>
                    </Link>

                    <Link to="/Login" style={{textDecoration:'none',color:'white'}}>
                        <Button color="inherit">Login</Button>
                    </Link>

                </Toolbar>
            </AppBar>
        </div>
    );
}
