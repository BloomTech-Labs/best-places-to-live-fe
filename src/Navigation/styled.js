import styled from "styled-components";

// export const Header = styled.div`
// background-color:white
// width: 20%;
// display: flex;
// align-items: center;
// justify-content: start-end;
// `;

export const NavBar = styled.nav.attrs({
    className: "fixed-nav"
})`
position:fixed;
background-color:white
width: 100%;
 display: flex;
 align-items: center;
 justify-content: start-end;
 padding: 0.5% 0% 0.5% 2%;
`;

export const LogoBox = styled.img`
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  position: relative;
  margin-right: 5px;
  width: 50px;
`;

export const NavBtn = styled.button`{
 font-weight: 600;
 text-align: center;
 padding: 8px 16px;
 border: 0px solid white;
 background: none;
 border-radius: 8px;
 cursor: pointer;
 font-family: ‘Rubik’, sans-serif;
 font-size: 1.5rem;
 outline:none; 
 text-decoration:none;
 &:hover {
   background-color: #E67271;
   color: white;
   border: none;
 }`;

export const Krunker = styled.a`
border;1px solid red;
position:absolute
left:97%;
color:white;
&:hover{
color:white;
}`;
