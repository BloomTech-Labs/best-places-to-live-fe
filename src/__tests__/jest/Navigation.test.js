import React from "react";
import Navigation from "../../Navigation";
import renderer from "react-test-renderer";
import routes from "../../app/routes";
import { Router, withRouter } from "react-router-dom";
const createHistory = require("history").createBrowserHistory;
const history = createHistory();

describe("<Navigation />", () => {
  it("Matches Snapshot", () => {
    const NavWithRouter = withRouter(routes);
    const tree = renderer
      .create(
        <Router history={history}>
          <NavWithRouter />
        </Router>
      )
      .toJSON();
    // @ts-ignore
    expect(tree).toMatchSnapshot();
  });
});
