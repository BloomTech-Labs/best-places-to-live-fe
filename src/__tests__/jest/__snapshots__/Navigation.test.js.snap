// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Matches Snapshot 1`] = `
<div>
  <nav
    className="sc-bdVaJa fixed-nav fClUDt"
    style={
      Object {
        "zIndex": "100",
      }
    }
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="sc-bwzfXH cpUCpO"
        src="LMHiconcopy.png"
      />
    </a>
    <a
      href="/Userlist"
      onClick={[Function]}
    >
      <button
        className="sc-htpNat esFguO"
      >
        Users
      </button>
    </a>
    <a
      href="/Maps"
      onClick={[Function]}
    >
      <button
        className="sc-htpNat esFguO"
        id="registerButton"
      >
        Maps
      </button>
    </a>
    <a
      href="/Login"
      onClick={[Function]}
    >
      <button
        className="sc-htpNat esFguO"
        id="loginButton"
      >
        Login
      </button>
    </a>
    <a
      href="/Register"
      onClick={[Function]}
    >
      <button
        className="sc-htpNat esFguO"
        id="registerButton"
      >
        SignUp
      </button>
    </a>
  </nav>
  <div
    className="sc-ifAKCX lpcuNw"
  >
    <span
      className="sc-EHOje geTijh"
    >
      <h1>
         Discover a place you'll love to live
      </h1>
      <form
        onSubmit={[Function]}
      >
        <input
          id="searchTerm"
          label="Enter City or State"
          onChange={[Function]}
          placeholder="Enter City or State"
          type="search"
          value=""
        />
      </form>
    </span>
  </div>
  <div
    className="sc-chPdSV isogWG"
  >
    <div
      className="sc-kgoBCf drgzer"
    >
      <h1>
        How we can help a Homie out
      </h1>
      <p>
        With over 5,000 cities available to explore our categories can help you easily find a place to call home. Use them to find the top scoring cities in the category that's most significant to you.
      </p>
    </div>
    <div
      className="sc-kGXeez Pgidq"
    >
      <div
        className="sc-kpOJdX hbvxaY"
      >
        <img
          className="sc-dxgOiQ idtcyj"
          src="budget.png"
        />
        <h4>
          Cost of Living
        </h4>
        <p>
          The amount of money needed to sustain a certain standard of living by affording basic expenses such as housing, food, taxes, and healthcare.
        </p>
      </div>
      <div
        className="sc-kpOJdX hbvxaY"
      >
        <img
          className="sc-dxgOiQ idtcyj"
          src="travel.png"
        />
        <h4>
          Commute
        </h4>
        <p>
          A regular journey of some distance such as to and from your place of work. How long will it take the homies to get where they want to go?
        </p>
      </div>
      <div
        className="sc-kpOJdX hbvxaY"
      >
        <img
          className="sc-dxgOiQ idtcyj"
          src="police-car.png"
        />
        <h4>
          Safety
        </h4>
        <p>
          A collection of overall crime rates in the area so all our homies can stay safe!
        </p>
      </div>
      <div
        className="sc-kpOJdX hbvxaY"
      >
        <img
          className="sc-dxgOiQ idtcyj"
          src="tolerance.png"
        />
        <h4>
          Tolerance
        </h4>
        <p>
          We respect homies of all races, religions, and orientations. Find a place to call home that does too!
        </p>
      </div>
    </div>
  </div>
  <div
    className="sc-htoDjs dNimok"
  >
    <div
      className="sc-dnqmqq kzrbag"
    >
      <h1>
        Let's move homie! Explore the places to call home.
      </h1>
      <p>
        Customize your search. Click on a category. Find a place you'll love.
      </p>
    </div>
    <div
      className="sc-gqjmRU hfUSCJ"
    >
      <img
        className="sc-VigVT gfIigm"
        onClick={[Function]}
        src="budget.png"
      />
      <img
        className="sc-VigVT gfIigm"
        onClick={[Function]}
        src="travel.png"
      />
      <img
        className="sc-VigVT gfIigm"
        onClick={[Function]}
        src="police-car.png"
      />
      <img
        className="sc-VigVT gfIigm"
        onClick={[Function]}
        src="tolerance.png"
      />
    </div>
    <div
      className="sc-iwsKbI FdyKJ"
    />
  </div>
</div>
`;
